.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,10 +1,24 @@
-###### Before submitting the PR, please make sure you do the following
+Closes https://github.com/metabase/metabase/pull/[issue number]
 
-- [ ] If you're attempting to fix a translation issue, please submit your changes to our [POEditor project](https://poeditor.com/join/project/ynjQmwSsGh) instead of opening a PR.
+### Description
 
-### Tests
+Describe the overall approach and the problem being solved.
+
+### How to verify
+
+Describe the steps to verify that the changes are working as expected.
+
+1. New question -> Sample Dataset -> ...
+2. ...
+
+#### Before & After Screenshots
+
+_Insert images here if sensible or remove the section_
+
+### Checklist
 
-- [ ] Run the frontend and Cypress end-to-end tests with `yarn lint && yarn test`)
-- [ ] If there are changes to the backend codebase, run the backend tests with `clojure -X:dev:test`
 - [ ] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
       (unless it's a tiny documentation change).
+- [ ] Tests for the new functionality have been added.
+- [ ] The new functionality complies with a11y.
+- [ ] If changes affect visualizations in any way add `visual` label.
```

Comment:
In my experience, checklists are helpful for about one week and then they become noise that everyone ignores.

Reply:
If we want to enforce things like testing and a11y, we should add automation or linting, etc. that enforces it - but obviously that's a lot harder (e.g. sometimes you're changing things like styling that can't/don't need to be tested, which is hard to detect)

Reply:
At my previous job, it was helpful for things that cannot be automated or are not yet automated. 
And here even though we had an agreement that no code changes should be merged without tests it somehow slipped through our fingers. I assume if a person forgets to include specs and then sees that a checkbox stating that specs are added should be checked, it will be a nice reminder. We can add a workflow that ensures checklist items are checked.

I removed all other not yet agreed or too optional items from the checklist for now